pkg/virt-controller/watch/vm.go

```diff
@@ -1113,6 +1115,16 @@ func isSetToStart(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineInstance)
 	}
 }
 
+func (c *VMController) cleanupRestartRequired(vm *virtv1.VirtualMachine) (*virtv1.VirtualMachine, error) {
+	vmConditionManager := controller.NewVirtualMachineConditionManager()
+	if vmConditionManager.HasCondition(vm, virtv1.VirtualMachineRestartRequired) {
+		vm = vm.DeepCopy()
```

Comment:
I don't see much of a point as we're returning a vm which won't have that condition already.

Reply:
Can you elaborate? Are you saying the `DeepCopy` is not required?

Note: This is here because the `vm` comes from the cache. Therefore if an update of the status fails the cached `vm` would not contain the condition but the `vm` in the cluster would still have the condition.

Reply:
afaik, normally, the observer will only see a change if you modify an existing value of a slice adding or deleting an element will not be observed.
Here RemoveCondition replaces the reference to a slice, so I highly doubt anything is being corrupted.

That's my thinking. That's why I think anywhere else in the code this specific part was never a concern.